### DISCUSSION TITLE
Sloshing tank

### DISCUSSION THREAD

**Discussion by pemeunier** [ProjectPhysX/FluidX3D#201] [General]
Hi,

Thank you for this impressive work.

Do you think that your implementation would be adapted to calculation of sloshing tanks ?
I tried with Moving boundaries and Surface extension but I couldn't make it stable.

Would it be possible to get a simple example such as a box half filled with water following a sinusoïdal motion ?

Kind regards

**Comment by ProjectPhysX** [comment]
Hi @pemeunier,

have a look at [this sample setup](https://github.com/ProjectPhysX/FluidX3D/blob/v2.17/src/setup.cpp#L1209), it is sloshing in a cubic domain. To drive the sloshing, the vomume force is changed in between simulation time steps; this change comes at no extra compute cost. You can adapt your setup from this sample.

Kind regards,
Moritz
